Sources/SQLKit/Query/SQLUnion.swift

```diff
@@ -0,0 +1,24 @@
+public struct SQLUnion: SQLExpression {
+    public let args: [SQLExpression]
+    public let all: Bool
+
+    public init(_ args: SQLExpression..., all: Bool = false) {
+        self.init(args, all: all)
+    }
+
+    public init(_ args: [SQLExpression], all: Bool = false) {
+        self.args = args
+        self.all = all
+    }
+
+    public func serialize(to serializer: inout SQLSerializer) {
+        let groups = args.map(SQLGroupExpression.init)
+        guard let first = groups.first else { return }
+        first.serialize(to: &serializer)
+        for arg in groups.dropFirst() {
+            serializer.write(all ? " UNION ALL " : " UNION ")
+            arg.serialize(to: &serializer)
+        }
+    }
```

Comment:
I recommend doing something like this for arg handling and serialization so that we 1) don't silently accept a nonsensical construction like `query.union()`, 2) we can more easily support additional syntax for databases that may have it (like PostgreSQL's `UNION | INTERSECT | EXCEPT [ALL | DISTINCT]` syntax), and 3) maintain general composability. (Oh, and 4) consistent use of `self.` for accessing properties.)

```suggestion
    public init(_ args: [SQLExpression], all: Bool = false) {
        precondition(!args.isEmpty, "Empty SQLUnions are not valid.") 
        self.args = args
        self.all = all
    }

    public func serialize(to serializer: inout SQLSerializer) {
        self.args
            .map { [SQLGroupExpression($0)] }
            .joined([SQLUnionJoiner(all: self.all)])
            .forEach { $0.serialize(to: &serializer) }
    }
}

fileprivate struct SQLUnionJoiner: SQLExpression {
    let all: Bool
    
    func serialize(to serializer: inout SQLSerializer) {
        serializer.write("UNION\(self.all ? " ALL" : "") ")
    }
```

Reply:
Sorry, I had to revert the `map/joined/forEach` part because it didn't actually compile and it wasn't immediately obvious how to fix it. I've kept the precondition and the `self.`.

Reply:
Sorry about that! That's what happens when I start writing code in GitHub comments without actually trying it 😅. I'll fix the errors and repost shortly.

Reply:
No worries, Compiler As A Service 🤣